WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -266,6 +266,9 @@ def post_results
       unless comp.results
         return render html: "<div class='container'><div class='alert alert-warning'>No results</div></div>".html_safe
       end
+      unless comp.results_posted.nil?
```

Comment:
There is no `Competition#results_posted` method. Competition has `results_posted_at` timestamp and a convenient `results_posted?` method: https://github.com/thewca/worldcubeassociation.org/blob/c89a14ae08fb8ad4815968f6c1929021d2cdda84/WcaOnRails/app/models/competition.rb#L873-L875
So the condition should be as straightforward as `if comp.results_posted?` =)